assets/javascripts/discourse/components/alternate-filter-name-picker.js

```diff
@@ -0,0 +1,29 @@
+import Component from "@glimmer/component";
+import { action } from "@ember/object";
+import { ajax } from "discourse/lib/ajax";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import { debounce } from "discourse-common/utils/decorators";
+
+export default class AlternateFilterNamePicker extends Component {
+  @debounce(500)
```

Comment:
Would be a considerable refactor, but it might be good idea to introduce a "Save" button in this view, that way we send only 1 request to the backend once everything looks good. This is an issue for the matrix as well, a request every box checked will cause unnecessary extra work. 

(It's not a blocker for this PR.)